exporter/loadbalancingexporter/resolver_k8s_handler.go

```diff
@@ -0,0 +1,102 @@
+package loadbalancingexporter
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var _ cache.ResourceEventHandler = (*handler)(nil)
+
+type handler struct {
+	endpoints *sync.Map
+	callback  func(ctx context.Context) ([]string, error)
+	logger    *zap.Logger
+}
+
+func (h handler) OnAdd(obj interface{}, isInInitialList bool) {
+	h.logger.Debug("onAdd called")
+	var endpoints []string
+
+	switch object := obj.(type) {
+	case *corev1.Endpoints:
+		endpoints = convertToEndpoints(object)
+	default: // unsupported
+		h.logger.Warn("onAdd unable to handle object", zap.Any("obj", obj))
+		return
+	}
+	changed := false
+	for _, ep := range endpoints {
+		if _, loaded := h.endpoints.LoadOrStore(ep, true); !loaded {
+			changed = true
+		}
+	}
+	h.logger.Debug("onAdd check", zap.Bool("changed", changed))
+	if changed {
+		h.callback(context.Background())
+	}
+}
+
+func (h handler) OnUpdate(oldObj, newObj interface{}) {
+	h.logger.Debug("onUpdate called")
+	switch oldObj.(type) {
+	case *corev1.Endpoints:
+		newEps, ok := newObj.(*corev1.Endpoints)
+		if !ok {
+			return
+		}
+		endpoints := convertToEndpoints(newEps)
+		changed := false
+		for _, ep := range endpoints {
+			if _, loaded := h.endpoints.LoadOrStore(ep, true); !loaded {
+				changed = true
+			}
+		}
+		h.logger.Debug("onUpdate check", zap.Bool("changed", changed))
+		if changed {
+			h.callback(context.Background())
+		}
+	default: // unsupported
+		h.logger.Warn("onUpdate unable to handle object", zap.Any("obj", oldObj))
```

Comment:
```suggestion
		h.logger.Warn("Got an unexpected Kubernetes data type during the update of the pods for a service", zap.Any("obj", oldObj))
```

Reply:
Thanks a lot, this makes the meaning very clear.